tests/cases/fourslash/genericCombinators3.ts

```diff
@@ -1,6 +1,8 @@
 /// <reference path='fourslash.ts'/>
 
 ////interface Collection<T, U> {
+////    dummyT: T;

```

Comment:
This is setting off breaking change alarm bells for me. What's the impact here?

Reply:
If a user (like this test case) only relies on inference to match up different parameterizations of _the same_ generic interface `Collection<T, U>` and never passes a subinterface or an implementing class where a `Collection<T, U>` is expected, then they may have gotten away (until this PR) with having no reference to `T` or `U` in `Collection<T, U>`.  After this PR, no inference will be made and their code will likely fail to type check until dummy properties referencing `T` and `U` are added, as I've done to the test case.  If a user expects inference to work from a subinterface or an implementing class to `Collection<T, U>`, then they would already have seen the problem and would already have been forced to add the dummy properties.

Does that answer your question?  Perhaps running the PR against your real-world code suite is the best way to get an idea of who's relying on the current inconsistent and undocumented behavior.

Reply:
So we'll start effectively requiring the use of structures similar to [phantom data](https://doc.rust-lang.org/core/marker/struct.PhantomData.html). At some point do we make an unreferenced type parameter as a proper error?

Reply:
> So we'll start effectively requiring the use of structures similar to phantom data.

As stated [in the FAQ](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-doesnt-type-inference-work-on-this-interface-interface-foot---), we _already_ require this in all cases except the one inconsistent case I'm fixing in this PR.

Reply:
This case is the only case that the PR changes, so if you don't want to change this case, I guess I'll abandon the PR.  I wish you had told me [here](https://github.com/Microsoft/TypeScript/issues/26815#issuecomment-417743235) before I did the work.